src/common/database.c

```diff
@@ -1854,6 +1854,15 @@ static int _upgrade_data_schema_step(dt_database_t *db, int version)
 
     new_version = 6;
   }
+  else if(version == 6)
+  {
+    TRY_EXEC("CREATE TABLE data.locations "
+             "(locid INTEGER PRIMARY KEY, type INTEGER, longitude REAL, latitude REAL, "
```

Comment:
Sorry I see that I was wrong, the id here is really the tagid to which it corresponds, right ?

If I detach a tag the picture is till in the location, the location still count this image as in (as shown by the count around parenthesis). So maybe something to improve here ?

Reply:
To make this clear we need to write:

`CREATE TABLE data.locations (tagid , locid INTEGER PRIMARY KEY, type INTEGER, longitude REAL, latitude REAL, delta1 REAL, delta2 REAL, FOREIGN KEY(tagid) REFERENCES tags(id));`

Reply:
> If I detach a tag the picture is till in the location. 

Just to be sure: you detach the tag in tagging plugin, correct ?

Reply:
Good catch!

This case should be avoided. The relationship is not user dependent but location data dependent.
If the user removes the tag, he screws the location mechanisms.
In the same way, it would be wrong to attach the location tag to an image outside the location area.
Unless it's intentional....
Not sure about we should do here...

Reply:
Additional note. As soon the location changes, the user action will be lost (as the relationship is calculated again).

Reply:
I see 2 possibilities.
1. forbid to attach/detach locations from tagging. This seems to be the more logical (as the relationship is calculated).
2. "signal" the change to location plugin, which in turn recalculates the location's images and undoes the user action. This may give inconsistent result between map and collect module (if collect module is triggered first).

Thought ?

Reply:
In fact "darktable" should not be attachable/detachable by a direct user action, as they are all darktable mechanism dependent.
Do you agree with that ?

Reply:
Yes, I would tend to agree that internal dt tags should not be manipulated in general. I'm not sure to see all the consequences if we forbid that but I'm ok to do that for 3.4. It cannot make dt unusable anyway just maybe some annoyances...